File: internal/domain/kyc.go
package domain

import "time"

const (
	KYCStatusPending  = "pending"
	KYCStatusApproved = "approved"
	KYCStatusRejected = "rejected"
)

type KYCVerification struct {
	ID              string     `json:"id"`
	UserID          string     `json:"user_id"`
	IDCardNumber    string     `json:"id_card_number"`
	IDCardImageURL  string     `json:"id_card_image_url"`
	SelfieImageURL  string     `json:"selfie_image_url"`
	Status          string     `json:"status"`
	RejectionReason *string    `json:"rejection_reason,omitempty"`
	SubmittedAt     time.Time  `json:"submitted_at"`
	VerifiedAt      *time.Time `json:"verified_at,omitempty"`
}

File: web/src/types/kyc.ts
export type KYCStatus = 'PENDING' | 'APPROVED' | 'REJECTED' | 'NOT_SUBMITTED';

export interface KYCResponse {
    id: string;
    user_id: string;
    id_card_number: string;
    id_card_image_url: string;
    selfie_image_url: string;
    status: KYCStatus;
    rejection_reason?: string;
    submitted_at: string;
    verified_at?: string;
}

export interface SubmitKYCRequest {
    id_card_number: string;
    id_card_image_url: string;
    selfie_image_url: string;
}
